File: android/src/main/java/com/mcodex/nitroarchive/infrastructure/memory/BufferArchiveInput.kt
package com.mcodex.nitroarchive.infrastructure.memory

import com.mcodex.nitroarchive.application.ports.ArchiveInput

class BufferArchiveInput(private val data: ByteArray) : ArchiveInput {
    override val estimatedSize: Long? get() = data.size.toLong()

    override suspend fun openForRandomAccess(): ByteArray = data
}

File: ios/Infrastructure/Memory/BufferArchiveInput.swift
import Foundation

final class BufferArchiveInput: ArchiveInput {
  private let data: Data
  var estimatedSize: UInt64? { UInt64(data.count) }

  init(data: Data) {
    self.data = data
  }

  func openForRandomAccess() async throws -> Data {
    return data
  }
}
